lib/vscode/src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts

```diff
@@ -406,10 +406,44 @@ export class ExtensionsViewPaneContainer extends ViewPaneContainer implements IE
 		this.root = parent;
 
 		const overlay = append(this.root, $('.overlay'));
+
 		const overlayBackgroundColor = this.getColor(SIDE_BAR_DRAG_AND_DROP_BACKGROUND) ?? '';
 		overlay.style.backgroundColor = overlayBackgroundColor;
 		hide(overlay);
 
+		// NOTE@coder this UI element helps users understand the extension marketplace divergence
+		const extensionHelperLocalStorageKey = 'coder.extension-help-message';
+
+		if (localStorage.getItem(extensionHelperLocalStorageKey) === null) {
+			const helperHeader = append(this.root, $('.header'));
+			helperHeader.id = 'codeServerMarketplaceHelper';
+			helperHeader.style.height = 'auto';
+			helperHeader.style.fontWeight = '600';
+			helperHeader.style.padding = 'padding: 5px 16px';
+			helperHeader.style.position = 'relative';
+			helperHeader.innerHTML = `
+			<div style="margin-bottom: 8px;">
+			<p style="margin-bottom: 0; display: flex; align-items: center"><span class="codicon codicon-warning" style="margin-right: 2px; color: #C4A103"></span>WARNING</p>
+			<p style="margin-top: 0; margin-bottom: 4px">
+			These extensions are not official. Find open-source extensions
+			<a href="https://open-vsx.org/" target="_blank">here</a>.
+			See <a href="https://github.com/cdr/code-server/blob/master/doc/FAQ.md#differences-compared-to-vs-code" target="_blank">docs</a>.
+			</p>
+			</div>
+					`;
+
+			const dismiss = append(helperHeader, $('span'));
+			dismiss.innerHTML = 'Dismiss';
+			dismiss.style.display = 'block';
+			dismiss.style.textAlign = 'right';
+			dismiss.style.cursor = 'pointer';
+			dismiss.onclick = () => {
+				// Eventually removes helperHeader from DOM
+				setTimeout(() => this.root?.removeChild(helperHeader));
```

Comment:
Why do we need the timeout?

Reply:
Oh also we could tweak this a bit so it doesn't depend on `this.root`. Probably fine either way but since `this.root` could be reassigned I guess it's technically safer this way:

```suggestion
				setTimeout(() => helperHeader.remove());
```

Reply:
> Why do we need the timeout?

Good question...I am not sure why I added that 🤔 I'll remove it.

Reply:
> Oh also we could tweak this a bit so it doesn't depend on `this.root`. Probably fine either way but since `this.root` could be reassigned I guess it's technically safer this way:
> 
> 
> 
> ```suggestion
> 
> 				setTimeout(() => helperHeader.remove());
> 
> ```

Great suggestion!